### ISSUE TITLE
Refactor CDR3-network construction. 

### ISSUE THREAD

**Issue by grst** [icbi-lab/scirpy#191]
Currently, we first build a `sequence` x `sequence` CDR3-similarity network. We then expand this network to a `cell` x `cell` similarity network. 

Using a `cell` x `cell` similarity network is suboptimal. If there are highly expanded clonotypes (i.e. cells with the same CDR3 sequences), this leads to huge blocks in the matrix, that cannot be efficiently stored as a sparse matrix. 

This issue proposes to get rid of the `cell` x `cell` matrix entirely and use the clonotype annotation + CDR3-similarity network directly instead. 

Supersedes #135. (no patching of the edges function will be required any more)

Affected functions: 
 - [ ] Plotting the clonotype network: Instead of blobs of individual cells, we would show connected bubbles. The size of the bubbles corresponds to the clonotype size. Multiple colors can be visualized as a pie chart. 
     * it should be possible to visualize the clonotype number even when not coloring by clonotype
 - [ ] tbc

Additional changes: 
- [ ] Currently, the `same_v_gene` and `same_receptor_type` options only work on a clonotype-definition level. It would be beneficial to offer these already on a network-level, i.e. that nodes with different v-genes are not connected within the network. 
- [ ] will affect how dual_tcr works. Good chance to take care of #164 (make dual_tcr more prominent)

### Notes

 * the trick to make this work is to move the `receptor_arms` and `dual_tcr` arguments to the `define_clonotypes` function, and the clonotype definition *before* the network construction.

**Comment by josinejansen** [comment]
Dear ICBI lab,

I have some questions regarding the Scirpy package, I would really appreciate your help.
For my analyses I first merged TCR and transcriptomics data from two different samples (organ 1 and organ 2), subsequently I then merged these two files.

**1. a) About the clonotype_network: as I understand, each node represents a cell, and edges connect cells belonging to the same clonotype. The function makes visualization of the clonotype-network possible, analogous to the construction of a neighborhood graph from transcriptomics data with the Scanpy package, so that based on the above, it computes a neighborhood graph of CDR3 nucleotide sequences with "scirpy.pp.tcr_neighbors())". My question is: are the clusters (where each node represents a cell) based on identical nucleotide sequences, or is it based on similarity, meaning that one cluster could consists of cells with slightly different nucleotide sequences?**

**b) Do the lines that connect the cells within a cluster have a meaning?**

<img width="528" alt="Screen Shot 2020-10-14 at 2 24 29 PM" src="https://user-images.githubusercontent.com/70770407/96029730-0f6a3a00-0e29-11eb-8165-d342e0018d23.png">

**c) In the plot above, is it correct that the closer the different clusters are together, the more similar their nucleotide sequences are? Meaning that the sequences of the clonotypes consisting of only 2 cells in this case are most different from the clonotype clusters consisting of >5 cells (as they are further apart)?**

**2. a) The package allows for specifying what organs the clusters consist of:**

<img width="544" alt="Screen Shot 2020-10-14 at 2 32 06 PM" src="https://user-images.githubusercontent.com/70770407/96033239-d41e3a00-0e2d-11eb-881e-ca4b3cd143af.png">

**My question is: how can I select specific clusters in the clonotype network graph (I only would like to include the clonotypes in the network that have identical clonotypes between the two different samples from my data, i.e. organ 1 and organ 2), meaning that in the plot above I would like to filter for the clusters only that display both blue and orange nodes.**

**b) How can I assign numbers to my clusters based on identical nucleotide sequences shared between two samples?**

**c) How can I add a legend to my plot, and how can I change the name 'batch' into 'sample'?**

**3. Using Scirpy, how can one best specify differentially expressed genes (based on the transcriptomics data) between the different clusters based on shared nucleotide sequences between blood and fat, for example _cluster 1 vs. rest of the clusters_, and _cluster 1 and 2 vs. cluster 3 and 4_? I can't make any code work, I have tried Scanpy's tool to rank genes using WIlcoxon, but unfortunately I can't make It work.

_sc.tl.rank_genes_groups(adata, 'clonotype', groups=['1','2'], reference=['3','4'], method='wilcoxon')
sc.pl.rank_genes_groups(adata, groups=['1', '2'], n_genes=20)_

-The key of the observations grouping to be considered would be: 
"clonotype clusters based on share identical nucleotide sequences between organ 1 and organ 2".
-Subset of groups to which comparison wouldl be restricted: "clonotype 1 and clonotype 2"
-Comparison: Compare with respect to a specific group
-Group identifier with respect to which compare: "clonotype 3 and clonotype 4"
“The number of genes that appear in the returned tables”: 100
“Method”: Wilcoxon-Rank-Sum

Thanks in advance,

Josine
